lib/commands/web.js

```diff
@@ -269,12 +273,62 @@ extensions.translateWebCoords = async function (coords) {
 };
 
 extensions.checkForAlert = async function () {
+  try {
+    const alertText = await this.proxyCommand('/alert/text', 'GET');
+    if (util.hasValue(alertText)) {
+      log.debug(`Found an alert with text '${alertText}'`);
+      return true;
+    }
+  } catch (ign) {}
   return false;
 };
 
 extensions.waitForAtom = async function (promise) {
-  // TODO: Add check for alert and accept/dismiss it as per autoAcceptAlert capability
-  return this.parseExecuteResponse(await promise);
+  let isWaitingForAtom = true;
+  const alertsHandler = async () => {
+    while (isWaitingForAtom) {
+      await B.delay(ALERT_CHECK_INTERVAL);
+      if (!isWaitingForAtom) {
+        return;
+      }
+
+      if (await this.checkForAlert()) {
+        if (this.opts.autoAcceptAlerts) {
+          log.debug('Accepting the alert as requested by autoAcceptAlerts capability');
+          try {
+            await this.proxyCommand('/alert/accept', 'POST', {});
+          } catch (err) {
+            log.warn(`Did not accept the alert. Original error: ${err.message}`);
+          }
+        } else if (this.opts.autoDismissAlerts) {
+          log.debug('Dismissing the alert as requested by autoDismissAlerts capability');
+          try {
+            await this.proxyCommand('/alert/dismiss', 'POST', {});
+          } catch (err) {
+            log.warn(`Did not dismiss the alert. Original error: ${err.message}`);
+          }
+        } else {
```

Comment:
Can this be moved out, so if neither `autoAcceptAlerts` nor `autoDismissAlerts` is on? So down on line 319 it would be something like
```js
if (!this.opts.autoAcceptAlert && !this.opts.autoDismissAlert) {
  alertsHandler().catch((e) => log.warn(e));
} else {
  log.debug('Doing nothing...');
}
```

Reply:
yep, I can change it this way.